pkg/repo/helm.go

```diff
@@ -25,6 +25,19 @@ func Write(repofile string, o *repo.Entry, helm *helm.EnvSettings) error {
 		return err
 	}
 
+	if _, err := os.Stat(repofile); os.IsNotExist(err) {
+		f, err := os.Create(repofile)
+		if err != nil {
+			return err
+		}
+		err = f.Close()
```

Comment:
`defer` is prefer

Reply:
you can't catch errors and return them via deferred functions